File: apps/etterlatte-pdltjenester/src/main/kotlin/personweb/personopplysninger/Personopplysninger.kt
package no.nav.etterlatte.personweb.dto

import no.nav.etterlatte.libs.common.person.Folkeregisteridentifikator
import no.nav.etterlatte.libs.common.person.Sivilstatus
import no.nav.etterlatte.libs.common.person.Utland
import no.nav.etterlatte.libs.common.person.VergemaalEllerFremtidsfullmakt
import java.time.LocalDate
import java.time.LocalDateTime

data class Personopplysninger(
    val soeker: PersonopplysningPerson?,
    val avdoede: List<PersonopplysningPerson?>,
    val gjenlevende: List<PersonopplysningPerson?>,
)

data class PersonopplysningPerson(
    val fornavn: String,
    val etternavn: String,
    val foedselsnummer: Folkeregisteridentifikator,
    val foedselsdato: LocalDate?,
    val doedsdato: LocalDate?,
    val bostedsadresse: List<Bostedsadresse>?,
    val sivilstand: List<Sivilstand>?,
    val statsborgerskap: String?,
    val pdlStatsborgerskap: List<PdlStatsborgerskap>?,
    val utland: Utland?,
    val familierelasjon: Familierelasjon?,
    val avdoedesBarn: List<PersonopplysningPerson?>?,
    val vergemaalEllerFremtidsfullmakt: List<VergemaalEllerFremtidsfullmakt>?,
)

data class Bostedsadresse(
    val adresse: String?,
    val postnr: String?,
    val gyldigFraOgMed: LocalDateTime?,
    val gyldigTilOgMed: LocalDateTime?,
    val aktiv: Boolean,
)

data class Sivilstand(
    val sivilstatus: Sivilstatus,
    val relatertVedSivilstand: Folkeregisteridentifikator?,
    val gyldigFraOgMed: LocalDate?,
)

data class PdlStatsborgerskap(
    val land: String,
    val gyldigFraOgMed: LocalDate?,
    val gyldigTilOgMed: LocalDate?,
)

data class Familierelasjon(
    val ansvarligeForeldre: List<Folkeregisteridentifikator>?,
    val barn: List<Folkeregisteridentifikator>?,
)

File: apps/etterlatte-saksbehandling-ui/client/src/shared/types/personopplysninger.ts
import { Utland } from '~shared/types/Person'
import { VergemaalEllerFremtidsfullmakt } from '~components/person/typer'

export interface Personopplysninger {
  soeker?: PersonopplysningPerson
  avdoede?: PersonopplysningPerson[]
  gjenlevende?: PersonopplysningPerson[]
}

export interface PersonopplysningPerson {
  fornavn: string
  etternavn: string
  foedselsnummer: string
  foedselsdato?: Date
  doedsdato?: Date
  bostedsadresse?: Bostedsadresse[]
  sivilstand?: Sivilstand[]
  statsborgerskap?: string
  pdlStatsborgerskap?: PdlStatsborgerskap[]
  utland?: Utland
  familierelasjon?: Familierelasjon
  avdoedesBarn?: PersonopplysningPerson[]
  vergemaalEllerFremtidsfullmakt?: VergemaalEllerFremtidsfullmakt[]
}

export interface Bostedsadresse {
  adresse?: string
  postnr?: string
  gyldigFraOgMed?: string
  gyldigTilOgMed?: string
  aktiv: boolean
}

export interface Sivilstand {
  sivilstatus: string
  relatertVedSivilstand?: string
  gyldigFraOgMed?: Date
}

export interface PdlStatsborgerskap {
  land: string
  gyldigFraOgMed?: string
  gyldigTilOgMed?: string
}

export interface Familierelasjon {
  ansvarligeForeldre?: string[]
  barn?: string[]
}
